### PR TITLE
Final project submition

### PR DESCRIPTION
**Pull request by Hombre2014**
This is the final project pull request where all the requirements for the project have been implemented as follow:

## Description
The JavaScript capstone project  is about building your own web application based on an external API. You will select an API that provides data about a topic that you like and then build the webapp around it.

## Video presentation
[Video link](https://youtu.be/QOhLsW7cuFU)

## Live Demo
[Live demo link](https://chamkoriyski.me/JS_Capstone_project/dist)

### APIs
- First, you need to find an API so you can base the development of the webapp around it. The API should allow you to:
  - Get a list of items with a unique item id (or generate the unique id).
  - For a given item, get detailed information about it.
- You will use our [Involvement API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270) to record the different user interactions (likes, comments and reservations).

### Interfaces
- You should build these interfaces:
  - The home page.
  - The comments popup.
- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Home page
  - When the page loads, the webapp retrieves data from:
    - The selected API and shows the list of items on screen.
    - The Involvement API to show the item likes.
  - Remember that your page should make only 2 requests:
    - One to the base API.
    - And one to the Involvement API.
  - When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
  - When the user clicks on the "Comments" button, the Comments popup appears.
  - Home page header and navigation similar to the given mockup.
  - Home page footer similar to the given mockup.
- Comments popup
  - When the popup loads, the webapp retrieves data from:
    - The selected API and shows details about the selected item.
    - The Involvement API to show the item comments.
  - When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.

### Counters
Counters We have counters in all the interfaces that show:
- The number of items (home).
- The number of comments (comments popup).

### Technical set up
- Set up the repository on GitHub and use Gitflow.
- Set up webpack.
- Set up a JavaScript testing library (Jest).

### CODE REVIEW

File: src/comments.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    method: 'POST',
+    headers: { 'Content-type': 'application/json' },
+    body: JSON.stringify({ item_id: `'${itemId}'`, username: userName, comment: userComment }),
+  })
+    .then((res) => res.ok)
+    .then((data) => (data));
+}
+
+export function displayComments(itemId) {
+  fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/1nLM5MTDuqVGBJxBgtuq/comments?item_id='${itemId}'`)
```
Comment (elisha2kyakpo1) [inline src/comments.js]: - [ ] Let's use ES6 syntax for functions, making use of the arrow functions

File: src/comments.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      }
+      const allComments = document.getElementsByClassName('single-comment');
+      const counter = allComments.length;
+      const commentsCounter = document.getElementsByClassName('comments-count');
+      commentsCounter[0].innerText = `(${counter})`;
+    });
+}
+
+export function getComments(itemId) {
+  fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/1nLM5MTDuqVGBJxBgtuq/comments?item_id='${itemId}'`)
```
Comment (elisha2kyakpo1) [inline src/comments.js]: - [ ]  Let's use ES6 syntax for functions, making use of the arrow functions.

File: src/commentsCounter.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export default function commentsCounter(movieComments) {
+  const commentsCounter = movieComments;
```
Comment (elisha2kyakpo1) [inline src/commentsCounter.js]: - [ ]  Let's use ES6 syntax for functions, making use of the arrow functions.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi team,

Great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight

- All the tests are passing✔✔
- The project uses async and await ✔✔
- The  project is almost complete✔✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- We should make sure the PR description highlights what was done in the project.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Hombre2014** [comment]
Thanks for the review and suggestions.
We have implemented all the changes as follow:
Changes `displayComments` function from normal to ES6 arrow standard.
Changes `getComments` function from normal to ES6 arrow standard.
Changes `commentsCounter` function from normal to ES6 arrow standard.

**Review by EricMbouwe — APPROVED** [review]
### Hi @team, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**